lib/improver/spotdata/extract_data.py

```diff
@@ -0,0 +1,580 @@
+           6. Convert back to temperature using the equations given above.
+           -------------------------------------------------------------------
+           ---------------------------RETURN RESULT---------------------------
+
+
+           ELSE: SpotData site height > model_surface --> Interpolate to site
+                                                          height.
+           -------------------------------------------------------------------
+
+           True surface above model surface (dz +ve) 'Unresolved hill'
+
+           ---upper_level--- Model level (k_upper)
+
+           ===site height=== SpotData site height
+
+           ---lower_level--- Model level (k_lower)
+
+           --model_surface-- Model level (k=0)
+
+
+           4. Use potential temperature gradient as an indicator of atmospheric
+              stability.
+
+              IF: dtheta/dz > 0 --> Stable atmosphere
+
+              ----------------------------------------------------------------
+              --> Stable
+              ----------------------------------------------------------------
+              5. Calculate pressure gradient between lower and upper model
+                 levels.
+              6. Use calculated gradients to extrapolate potential temperature
+                 and pressure to the SpotData site height.
+              7. Convert back to temperature using the equations given above.
+              -------------------------RETURN RESULT--------------------------
+
+
+              ELSE: dtheta/dz <= 0 --> Neutral/well-mixed atmosphere
+
+              ----------------------------------------------------------------
+              --> Neutral/well-mixed
+              ----------------------------------------------------------------
+              5. Use potential temperature from surface level; for a well mixed
+                 atmosphere dtheta/dz should be nearly constant.
+              6. Convert back to temperature using the pressure interpolated to
+                 SpotData site height.
+              -------------------------RETURN RESULT--------------------------
+
+
+        Args:
+        -----
+
+        cube : iris.cube.Cube
+            A cube of screen level temperatures at a single time.
+
+        sites/neighbours : See process() above.
+
+        surface_pressure : iris.cube.Cube
+            Cube of surface pressures at an equivalent time to the cube of
+            screen level temperatures.
+
+        lower/upper_pressure : iris.cube.Cube
+            Cubes of pressure data at the defined lower and upper model
+            levels, each at an equivalent time to the cube of screen level
+            temperatures.
+
+        lower/upper_temperature : iris.cube.Cube
+            Cubes of temperature data at the defined lower and upper model
+            levels, each at an equivalent time to the cube of screen level
+            temperatures.
+
+        dz_tolerance : float
```

Comment:
For dz_tolerance, dtheta_threshold, dz_max_adjustment: could you add the units and an explanation for the default values you've added.